SparkPost/matchbox
# [4.0.0] Refreshes global CSS, Adds Box and ThemeProvider components

### What Changed
This PR includes a lot of foundational work for the upcoming redesign. Pre-hibana stories are broken (visually), which is fine. The plan is to refactor these one by one.

- Installs `styled-components`, `styled-system`, and a complete set of `design-tokens`
- Maps tokens into styled system's theme via [theme.js](https://github.com/SparkPost/matchbox/pull/270/files#diff-1b2d2a590d7d8af96ecb27c45dca5b45)
- Refactors global CSS to match defaults for the redesign
- Adds a `Box` component
- Adds a `ThemeProvider` component

The `<Box/>` component is meant to be a generic component with access to every styled-system prop. Its goal is to reduce the amount of ad-hoc CSS in the UI, and make custom component styling easier to manage with direct access to design tokens.

#### Helpful links
- [Styled Components](https://www.styled-components.com/docs)
- [Styled System](https://styled-system.com/)
- [Styled System API](https://styled-system.com/api)
- [Styled System Theme Reference](https://styled-system.com/table)
- [SparkPost Design Tokens](https://github.com/SparkPost/design-tokens/blob/1.0.0-beta/dist/index.common.js)

### How To Test or Verify
- Run `npm run clean` and `npm i` to ensure dependancies are updated
- Run storybook with `npm run start:storybook`
- Visit http://localhost:9001/?path=/story/layout-box--styled-box and other stories for the Box component
- Or visit https://codesandbox.io/s/box-test-lyp6i
- Play around with props
- Provide any feedback around the system prop API or token naming conventions

---

<details>
<summary>Screenshots</summary>

![Screen Shot 2019-10-31 at 10 35 21 AM](https://user-images.githubusercontent.com/3903325/67956277-3274ed00-fbca-11e9-8303-405e60e39f74.png)
![Screen Shot 2019-10-31 at 10 35 14 AM](https://user-images.githubusercontent.com/3903325/67956291-3c96eb80-fbca-11e9-88fd-333a0fdf24ef.png)
</details>

<details>
<summary>PR Checklist</summary>

</details>
